apiserver/cloud/instance_information.go

```diff
@@ -75,6 +75,10 @@ func instanceTypes(api *CloudAPI,
 			result[i] = params.InstanceTypesResult{Error: common.ServerError(errors.NotValidf("asking %s cloud information to %s cloud", cloudTag.Id(), m.Cloud()))}
 			continue
 		}
+		if cons.CloudRegion == "" {
```

Comment:
Not all clouds have regions. I'm not sure if the validation belongs here or not, but if we do validate, we need to check that the cloud supports regions before requiring that it's empty.